services/ui-src/src/components/reports/ReportPage.tsx

```diff
@@ -76,6 +78,20 @@ export const ReportPage = ({ route }: Props) => {
     }
   };
 
+  const getEntities = (formEntityType: PageEntityType) => {
+    const entityTypeMap = {
+      plan: "aap-1", // reportData fieldId of the entity array
+      bssEntity: "absse-a8",
+      qualityMeasures: "apoc-3", // To Be Added
+      someOtherEntity: "apoc-4", // To Be Added
+    };
+    const pageEntityType: keyof typeof entityTypeMap = formEntityType;
+    const entitiesToFetch = entityTypeMap[pageEntityType];
+    const entities: string[] | undefined =
+      reportData?.fieldData[entitiesToFetch];
+    return entities;
+  };
```

Comment:
Because we're just passing all this stuff to the Entity section anyway, we should just pull in `reportData` to the entity section and do this all there rather than pass it through from `ReportPage`.

Reply:
I thought this too, but then I wasn't sure if we also wanted to use it for the `DynamicDrawerSection` so I moved it up to the `ReportPage`. I can stick it in the `StaticDrawerSection` then and do there if we aren't going to use it in both places